IdentityCore/src/cache/key/MSIDAppMetadataCacheQuery.m

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDAppMetadataCacheQuery.h"
+
+@implementation MSIDAppMetadataCacheQuery
+
+
+- (NSNumber *)type
+{
+    if (self.generalType != MSIDGeneralTypeOther)
+    {
+        return [super type];
+    }
+    
+    return nil;
+}
+
+- (NSString *)service
+{
+    if (self.clientId && (self.generalType != MSIDGeneralTypeOther))
```

Comment:
why do we need this check? (self.generalType != MSIDGeneralTypeOther)

Reply:
This is so so that we only look for app metadata types in the keychain.

Reply:
ok, then you should be checking that self.generalType == MsidAppMetadataType. Additionally, if both service and account are provided, it guarantees uniqueness of that entry, so I'm still not sure why this check is necessary. Let's discuss offline :)

Reply:
As discussed., kept the check as we still need it but changed it to check against MSIDAppMetadataType specifically.